eng/pipelines/templates/stages/archetype-ios-release.yml

```diff
@@ -51,13 +51,17 @@ stages:
               - template: ../variables/globals.yml
             pool:
               name: Azure Pipelines
-              vmImage: $(OSVmImage)
+              vmImage: macOS-10.15
             strategy:
               runOnce:
                 deploy:
                   steps:
                     - script: |
-                        echo "Pretend we are publishing to CocoaPods trunk.""
+                        pod trunk me
+                        pod trunk push $(System.DefaultWorkingDirectory)/sdk/${{parameters.ServiceDirectory}}/${{artifact.name}}/${{artifact.name}}.podspec.json --synchronous --use-modular-headers
```

Comment:
Are we checking elsewhere that the podspec in question has not already been published?

Reply:
Not explicitly.

Reply:
I don't know what the sequence for tagging/pushing to trunk is, but we should probably do that lest we tag a release and then find out we need additional commits because the version was not bumped. Then again, I suppose we could check that the tag doesn't already exist.

Reply:
The tag release process has a mechanism which triggers a failure if you try to tag a release again using a different commit sha (this is consistent across the engineering system). As part of that mechanism we actually check to see if that particular package has also been published (but its not quite explicitly what you are asking).

If we tag and fail to publish, and have to add more commits we have control of our GitHub releases and can delete the release/tag and then run the release process again (we do it all the time). Once things hit something we don't control (e.g. CocoaPods) then the version number is burnt.

For this reason, when we get around to doing the SPM sync (next) it might pay to do that before the CocoaPods publish.